File: leetcode/Python/find-winner-on-a-tic-tac-toe-game.py
class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        filled, turn, decision = 0, 0, 0
        square = [[0 for _ in range(3)] for _ in range(3)]
        score = [[0 for _ in range(8)] for _ in range(2)]
        for m in moves:
            square[m[0]][m[1]]=turn
            score[turn][m[0]]+=1
            score[turn][3+m[1]]+=1
            if m[0]==m[1]:
                score[turn][6]+=1
            if m[0]+m[1]==2:
                score[turn][7]+=1
            if score[turn][m[0]]==3 or score[turn][3+m[1]]==3 or score[turn][6]==3 or score[turn][7]==3:
                decision+=1
                break
            turn, filled = turn^1, filled+1
        if decision == 1:
            return 'A' if turn==0 else 'B'
        return 'Draw' if filled==9 else 'Pending'

File: leetcode/C++/find-winner-on-a-tic-tac-toe-game.cpp
class Solution {
public:
    string tictactoe(vector<vector<int>>& moves) {
        int filled=0, turn=0, decision=0;
        vector<vector<int>> square(3,vector<int>(3,0));
        vector<vector<int>> score(2,vector<int>(8,0));
        for(vector<int> &v : moves){
            square[v[0]][v[1]]=turn;
            if(++score[turn][v[0]]==3 || ++score[turn][3+v[1]]==3) ++decision;
            if(v[0]==v[1]&&++score[turn][6]==3) ++decision;
            if(v[0]+v[1]==2&&++score[turn][7]==3) ++decision;
            if(decision) break;
            turn^=1; ++filled;
        }
        if(decision) return (turn==0)? "A" : "B";
        return (filled==9)? "Draw" : "Pending";
    }
};